op-bindings/bindgen/remote_handlers.go

```diff
@@ -221,6 +241,41 @@ func (generator *bindGenGeneratorRemote) compareBytecodeWithOp(contractMetadataE
 	return nil
 }
 
+func (generator *bindGenGeneratorRemote) compareDeployedBytecodeWithRpc(contractMetadata *remoteContractMetadata, chain string) error {
```

Comment:
The code on this PR looks good, but I don't see any tests—are those added in a separate PR in the stack?

Reply:
No tests, just manual running of the code. For testing would simple unit tests suffice, or would you want to see actual requests to Etherscan/RPC?

Reply:
I'm not sure how we typically test code like this and I want to be careful not to introduce too much burden. But I think we should have unit tests, and a few integration tests with actual Etherscan/RPC calls, so we can test both the success cases, failure cases, and edge cases.

Without any tests it becomes hard to safely maintain and improve the tooling, and since the failure modes analysis shows that most issues require a hard fork or restarting the chain, we want to make sure we have confidence in the tooling we're using to mitigate those issues :)

Reply:
The actual requests to Etherscan/RPC node only happen upon manual running of `bindgen` (our tests run `bingen` for locally sourced contracts). Given that we're not going to be adding predeploys very frequently (and `bindgen` really only needs to be ran when we add new predeploys), there's probably going to be long periods of time `bindgen` isn't ran for remotely sourced contracts, and it could break and easily go unnoticed for a while - all this to say that I agree, we should test both unit and e2e

Reply:
I'll be adding these tests in a new PR as to not hold this one up any longer

Reply:
That works for me, thanks! Let me know where we should track adding tests then I'll mark this resolved + approve

Reply:
Added it as action item in the [failure mode analysis](https://www.notion.so/oplabs/Predeploy-Additions-Failure-Modes-and-Recovery-Path-Analysis-a5cfc284a11a4ae3b873399f02f3c966?pvs=4#250dc5296dd84689a1e3efed59497490)